common/src/main/java/com/wynntils/core/chat/tabs/ChatTabModel.java

```diff
@@ -41,6 +43,14 @@ public static void onWorldStateChange(WorldStateEvent event) {
         }
     }
 
+    @SubscribeEvent
+    public static void onScreenOpened(ScreenOpenedEvent event) {
+        if (!(event.getScreen() instanceof ChatScreen chatScreen)) return;
+        if (focusedTab == null || focusedTab.getAutoCommand() == null) return;
+
+        chatScreen.insertText(focusedTab.getAutoCommand(), true);
```

Comment:
Does this mean that if you switch between the tabs, the different auto commands will pop up in the chat message bar, and then be replaced with a new one when you switch to a new tab, and be removed if you select "local" etc? What if you have started to type a message and switch tab?

Reply:
Then the message will get replaced. I am not fully sure what legacy does, but imo it is fine this way.

Reply:
Eww... I dunno, really. :( Just removing what a player has typed is Not Okay, I think. That's the kind of user-generated data destruction behavior you can kind of kill your computer for.

Can't you check if anything has been typed, apart from another auto command, and if so, keep it?